src/db_adapters/access_keys.rs

```diff
@@ -36,53 +37,46 @@ pub(crate) async fn handle_access_keys(
-                            .get_mut(&(public_key.to_string(), receipt.receiver_id.to_string()))
-                        {
-                            existing_record.created_by_receipt_id =
-                                Some(receipt.receipt_id.to_string());
-                            existing_record.deleted_by_receipt_id = None;
-                        } else {
-                            access_keys.insert(
-                                (public_key.to_string(), receipt.receiver_id.to_string()),
-                                models::access_keys::AccessKey::from_action_view(
-                                    public_key,
-                                    &receipt.receiver_id,
-                                    access_key,
-                                    &receipt.receipt_id,
-                                ),
-                            );
-                        }
+                        access_keys.insert(
+                            (public_key.to_string(), receipt.receiver_id.to_string()),
+                            models::access_keys::AccessKey::from_action_view(
+                                public_key,
+                                &receipt.receiver_id,
+                                access_key,
+                                &receipt.receipt_id,
+                            ),
+                        );
                     }
                     near_primitives::views::ActionView::DeleteKey { public_key } => {
-                        if let Some(existing_record) = access_keys
-                            .get_mut(&(public_key.to_string(), receipt.receiver_id.to_string()))
-                        {
-                            existing_record.deleted_by_receipt_id =
-                                Some(receipt.receipt_id.to_string());
-                        } else {
-                            access_keys.insert(
-                                (public_key.to_string(), receipt.receiver_id.to_string()),
-                                models::access_keys::AccessKey {
-                                    public_key: public_key.to_string(),
-                                    account_id: receipt.receiver_id.to_string(),
-                                    created_by_receipt_id: None,
-                                    deleted_by_receipt_id: Some(receipt.receipt_id.to_string()),
-                                    permission: models::enums::AccessKeyPermission::NotApplicable,
-                                },
-                            );
-                        }
+                        access_keys.insert(
+                            (public_key.to_string(), receipt.receiver_id.to_string()),
+                            models::access_keys::AccessKey {
+                                public_key: public_key.to_string(),
+                                account_id: receipt.receiver_id.to_string(),
+                                created_by_receipt_id: None,
```

Comment:
We want to preserve the `created_by_receipt_id` if we had this access_key just added (thus I used `and_modify` in my comment)